sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -161,6 +162,10 @@ case class SortMergeJoinExec(
     sqlContext.conf.sortMergeJoinExecBufferInMemoryThreshold
   }
 
+  private def needEagerCleanup: Boolean = {
+    sqlContext.conf.getConf(SORT_MERGE_JOIN_EXEC_EAGER_CLEANUP_RESOURCES)
```

Comment:
Seems this only controls cleanup behavior in codege path?

Reply:
Ah yes, thanks for reminding, done in 631f3cb